src/sdl2_compat.c

```diff
@@ -1919,15 +1919,19 @@ EventFilter3to2(void *userdata, SDL_Event *event3)
         case SDL_EVENT_WINDOW_FOCUS_GAINED:
         case SDL_EVENT_WINDOW_FOCUS_LOST:
         case SDL_EVENT_WINDOW_CLOSE_REQUESTED:
-        case SDL_EVENT_WINDOW_TAKE_FOCUS:
         case SDL_EVENT_WINDOW_HIT_TEST:
         case SDL_EVENT_WINDOW_ICCPROF_CHANGED:
         case SDL_EVENT_WINDOW_DISPLAY_CHANGED:
             if (SDL3_EventEnabled(SDL2_WINDOWEVENT)) {
+                /* SDL_EVENT_WINDOW_TAKE_FOCUS (formerly event 528, now SDL_EVENT_WINDOW_HIT_TEST) was removed from SDL3.
+                 * Adjust the mapping offset when necessary.
+                 */
+                const Uint32 base_event = SDL_EVENT_WINDOW_SHOWN - (event3->type >= SDL_EVENT_WINDOW_HIT_TEST ? 1 : 0);
+
```

Comment:
Can you make this a function? Performance isn't a huge issue here, and I guarantee someone will get this wrong in the future.

Reply:
Done.